src/tests/test_kube_api.py

```diff
@@ -83,14 +75,344 @@ def _get_vips(cls, cluster_config, nodes: Nodes):
+        #  (please see cluster.get_primary_machine_cidr())
+        machine_cidr = nodes.controller.get_primary_machine_cidr()
+
+        agent_cluster_install = AgentClusterInstall(
+            api_client, f"{cluster_name}-agent-cluster-install", spoke_namespace
+        )
+
+        secret = Secret(api_client, f"{cluster_name}-secret", spoke_namespace)
+        secret.create(pull_secret=cluster_config.pull_secret)
+
+        cluster_deployment = ClusterDeployment(api_client, cluster_name, spoke_namespace)
+        cluster_deployment.create(agent_cluster_install_ref=agent_cluster_install.ref, secret=secret)
+
+        agent_cluster_install.create(
+            cluster_deployment_ref=cluster_deployment.ref,
+            image_set_ref=self.deploy_image_set(cluster_name, api_client),
+            cluster_cidr=cluster_config.cluster_networks[0].cidr,
+            host_prefix=cluster_config.cluster_networks[0].host_prefix,
+            service_network=cluster_config.service_networks[0].cidr,
+            ssh_pub_key=cluster_config.ssh_public_key,
+            hyperthreading=cluster_config.hyperthreading,
+            control_plane_agents=nodes.masters_count,
+            worker_agents=nodes.workers_count,
+            machine_cidr=machine_cidr,
+        )
+        agent_cluster_install.wait_to_be_ready(False)
+
+        if is_disconnected:
+            log.info("getting igntion and install config override for disconected install")
+            ca_bundle = self.get_ca_bundle_from_hub()
+            self.patch_install_config_with_ca_bundle(cluster_deployment, ca_bundle)
+            ignition_config_override = self.get_ignition_config_override(ca_bundle)
+        else:
+            ignition_config_override = None
+
+        proxy = self.setup_proxy(cluster_config, machine_cidr, cluster_name, proxy_server)
+
+        infra_env = InfraEnv(api_client, f"{cluster_name}-infra-env", global_variables.spoke_namespace)
+        infra_env.create(
+            cluster_deployment, secret, proxy, ignition_config_override, ssh_pub_key=cluster_config.ssh_public_key
+        )
+        agents = self.start_nodes(nodes, infra_env, cluster_config, is_ipv4)
+
+        if len(nodes) == 1:
+            self.set_single_node_ip(cluster_deployment, nodes, is_ipv4)
+
+        log.info("Waiting for agent status verification")
+        Agent.wait_for_agents_to_install(agents)
+
+        agent_cluster_install.wait_to_be_ready(True)
+
+        log.info("waiting for agent-cluster-install to be in installing state")
+        agent_cluster_install.wait_to_be_installing()
+
+        try:
+            log.info("installation started, waiting for completion")
+            agent_cluster_install.wait_to_be_installed()
+            log.info("installation completed successfully")
+        except Exception:
+            log.exception("Failure during kube-api installation flow:")
+            collect_debug_info_from_cluster(cluster_deployment, agent_cluster_install)
```

Comment:
hmm, why we don't raise the exception?

Reply:
Not sure. I kept the current flow

Reply:
I'd suggest raising the exception in that case
Recent events just taught us the severity of failing silently :joy:

Reply:
Lets do it on a different PR? I don't want that capi or late-binding will fail in this PR